### PR TITLE
Add a basic test for mosquitto kustomization

### PR DESCRIPTION
**Pull request by sdwilsh**

### CODE REVIEW

File: kustomization/tests/mosquitto/tasks.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (sdwilsh) [inline kustomization/tests/mosquitto/tasks.yml:1]: Whoops, this file shouldn't be here!

File: ansible/galaxy.yml [hunk 2]
```diff
@@ -26,7 +26,8 @@ tags:
 # collection label 'namespace.name'. The value is a version range
 # L(specifiers,https://python-semanticversion.readthedocs.io/en/latest/#requirement-specification). Multiple version
 # range specifiers can be set and are separated by ','
-dependencies: {}
+dependencies:
+  "kubernetes.core": ">=2.0.0"
```
Comment (robarnold) [inline ansible/galaxy.yml:30]: Do you not want to pin an exact version & bump w/renovate?

File: ansible/roles/minikube_test/vars/main.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+# vars file for minikube_test
+
+test_dir:
```
Comment (robarnold) [inline ansible/roles/minikube_test/vars/main.yml:4]: documentation?

File: Earthfile [hunk 4]
```diff
@@ -1,14 +1,86 @@
...
+#
+
+kustomization-tests-image:
+    # renovate: datasource=docker depName=python versioning=docker
+    ARG PYTHON_VERSION=3.12.0-bullseye
+    FROM python:$PYTHON_VERSION
+
+    # renovate: datasource=python depName=ansible-core
+    ARG ANSIBLE_VERSION=2.16.0
+    RUN python3 -m pip install ansible-core==$ANSIBLE_VERSION
```
[the remark this answers is not in the document]
Comment (sdwilsh) [reply Earthfile:49]: It does not use `requirements.txt`: https://github.com/marinatedconcrete/config/blob/main/ansible/Earthfile#L10-L12

But we can reduce the number of places we define this and ensure we're using the same versions.

File: ansible/galaxy.yml (hunk 2 above)
Comment (sdwilsh) [reply ansible/galaxy.yml:30]: I'm of two minds here.  While it's useful for us here to pin the version, it's not how you would typically releases these.  I would like to use it in my repo as well, and I have other roles/collections that might then conflict with a pinned version.

**Pull request merged**